### PR TITLE
migrate/deprecate discordapp.com => discord.com

### LINKED ISSUE #180: new discord API
- discordapp.com is being replaced with discord.com on 2020-11-07
  - https://discord.com/developers/docs/reference implies https://discord.com/api/v6 rather than https://discordapp.com/api/v7
- cdn.discordapp.com is remaining as-is though

### PR DESCRIPTION
**Pull request by casperdcl**
- [x] replace discordapp.com with discord.com (ahead of on 2020-11-07 move. cdn.discordapp.com is remaining as-is though)
- [x] confirm that everything still works?
- fixes #180

### CODE REVIEW

File: disco/api/http.py [hunk 1]
```diff
@@ -189,7 +189,7 @@ class HTTPClient(LoggingClass):
     A simple HTTP client which wraps the requests library, adding support for
     Discords rate-limit headers, authorization, and request/response validation.
     """
-    BASE_URL = 'https://discordapp.com/api/v7'
+    BASE_URL = 'https://discord.com/api/v6'
```
Comment (b1naryth1ef) [inline disco/api/http.py:192]: For now lets leave this as v7. I've confirmed with Discord staff that they don't currently intend on removing this version (which introduced better response payloads for errors) despite its lack of "official" support.

Comment (casperdcl) [reply disco/api/http.py:192]: right; there was no mention of it in their docs. Much clandestine such fun.

```suggestion
    BASE_URL = 'https://discord.com/api/v7'
```

**Comment by b1naryth1ef** [comment]
Thanks for getting this done :smiley_cat:, I'll work on getting this out in patch releases so folks can start migrating in preparation.

**Comment by casperdcl** [comment]
was minor since the API hasn't changed (was worried about the v7 -> v6 thing)

**Pull request merged**